build-locally.sh

```diff
@@ -213,25 +213,71 @@ function cleaning_up_before_we_start {
 }
 
 #-------------------------------------------------------------
+function get_architecture() {
+    h2 "Retrieving system architecture."
+        raw_os=$(uname -s) # eg: "Darwin"
+    os=""
+    case $raw_os in
+        Darwin*)
+            os="darwin"
+            ;;
+        Linux*)
+            os="linux"
+            ;;
+        *)
+            error "Unsupported operating system is in use. $raw_os"
+            exit 1
+    esac
+
+    architecture=$(uname -m)
+    case $architecture in
+        aarch64)
+            architecture="arm64"
+            ;;
+        amd64)
+            architecture="x86_64"
+    esac
+}
+
 function generate_beans {
     check_openapi2beans_is_installed
 
     h2 "Generating beans in the dev.framework.api.beans project."
-    cmd="openapi2beans generate --output ${BASEDIR}/galasa-parent/dev.galasa.framework.api.beans/src/main/java \
+    cmd="build/openapi2beans generate --output ${BASEDIR}/galasa-parent/dev.galasa.framework.api.beans/src/main/java \
     --yaml $BASEDIR/galasa-parent/dev.galasa.framework.api.openapi/src/main/resources/openapi.yaml \
-    --package dev.galasa.framework.api.beans"
+    --package dev.galasa.framework.api.beans.generated \
+    --force"
     $cmd 
     rc=$? ; if [[ "${rc}" != "0" ]]; then error "Failed to generate the api beans" ; exit 1 ; fi
     success "OK"
 }
 
 function check_openapi2beans_is_installed {
     h2 "Checking the openapi2beans tool is installed."
-    which openapi2beans > /dev/null
-    rc=$?; if [[ "$rc" != "0" ]]; then error "You don't have the openapi2beans tool installed. Go get it from the buildutils project" ; exit 1 ; fi
+    OPENAPI2BEANS_FOLDER="${BASEDIR}/galasa-parent/build"
+    if [[ ! -f ${OPENAPI2BEANS_FOLDER}/openapi2beans ]]; then
+        download_openapi2beans
+    fi
+    if [[ ! -x ${OPENAPI2BEANS_FOLDER}/openapi2beans ]]; then
+        chmod 700 ${OPENAPI2BEANS_FOLDER}/openapi2beans
```

Comment:
Can you replace `OPENAPI2BEANS_FOLDER` with `OPENAPI2BEANS` and includ the `/openapi2beans` as part of the variable value?
I can't see a reason why that would need to be separate and it makes for an easier to read code.

Reply:
ah that would be because there's an instance of just the folder being used, so it kinda made sense to do that instead, although the majority do use the openapi2beans path on the end, maybe I should have 2 constants instead?

Reply:
I'm happy to do it, just wanted an opinion on what might be the best course of action here